gradient/commands/datasets.py

```diff
@@ -561,141 +561,139 @@ def update_status():
-                            # tasks dispatched to the workers in the pool.
-                            # Since it's more of a PITA to properly distribute
-                            # this MPU among all workers than I really want to
-                            # deal with, that means we can't easily plug into
-                            # Halo for these updates. But we can print to
-                            # console! Which again, jank and noisy, but arguably
-                            # better than a task sitting forever, never either
-                            # completing or emitting an error message.
-                            if len(parts) % 7 == 0:  # About every 100MB
-                                print(
-                                    f'\nUploaded {len(parts) * part_minsize / 10e5}MB '
-                                    f'of {int(size / 10e5)}MB for '
-                                    f'{path}'
-                                )
-
-                    r = api_client.post(
-                        url=mpu_url,
-                        json={
-                            'datasetId': dataset_id,
-                            'version': version,
-                            'calls': [{
-                                'method': 'completeMultipartUpload',
-                                'params': {
-                                    'Key': key,
-                                    'UploadId': mpu_data['UploadId'],
-                                    'MultipartUpload': {'Parts': parts}
-                                }
-                            }]
-                        }
-                    )
-
-                self.validate_s3_response(r)
-            except requests.exceptions.ConnectionError as e:
-                return self.report_connection_error(e)
-            except Exception as e:
-                return e
+        try:
+            if size <= 0:
+                headers.update({'Content-Size': '0'})
+                r = session.put(url, data='', headers=headers, timeout=5)
+            # for files under 15MB
+            elif size <= (15e6):
+                with open(path, 'rb') as f:
+                    r = session.put(
+                        url, data=f, headers=headers, timeout=300)
+            # # for chonky files, use a multipart upload
+            else:
+                # Chunks need to be at least 5MB or AWS throws an
+                # EntityTooSmall error; we'll arbitrarily choose a
+                # 15MB chunksize
+                #
+                # Note also that AWS limits the max number of chunks
+                # in a multipart upload to 10000, so this setting
+                # currently enforces a hard limit on 150GB per file.
+                #
+                # We can dynamically assign a larger part size if needed,
+                # but for the majority of use cases we should be fine
+                # as-is
+                part_minsize = int(15e6)
```

Comment:
might be worth moving this byte size to a 'constant'. i see it referenced above too.